jaeger-core/src/test/java/io/jaegertracing/internal/JaegerSubclassTest.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (c) 2018, Uber Technologies, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.jaegertracing.internal;
+
+import io.jaegertracing.Configuration;
+import io.opentracing.Scope;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class JaegerSubclassTest {
+  private static class CustomConfiguration extends Configuration {
+    private CustomConfiguration(String serviceName) {
+      super(serviceName);
+    }
+
+    @Override
+    public CustomTracer.CustomBuilder getTracerBuilder() {
+      return (CustomTracer.CustomBuilder) super.getTracerBuilder();
+    }
+
+    @Override
+    public synchronized CustomTracer getTracer() {
+      return (CustomTracer) super.getTracer();
+    }
+
+    @Override
+    protected CustomTracer.CustomBuilder createTracerBuilder(String serviceName) {
+      return new CustomTracer.CustomBuilder(serviceName);
+    }
+  }
+
+  private static class CustomTracer extends JaegerTracer {
+    public static class CustomBuilder extends JaegerTracer.Builder {
+      private CustomBuilder(String serviceName) {
+        super(serviceName, new CustomObjectFactory());
+      }
+
+      @Override
+      public CustomTracer build() {
+        return (CustomTracer) super.build();
+      }
+
+      @Override
+      protected JaegerTracer createTracer() {
+        return new CustomTracer(this);
```

Comment:
neat! Managed without making builder fields public/protected :-)

Reply:
Ya I forgot this feature to, but it worked.